practical-unit-testing-with-rtl/Exercise2/problem/src/hooks/useTicTacToe.ts

```diff
@@ -0,0 +1,64 @@
+import { TicTacToeBoard, BoardSquareValue } from "@utilities/ticTacToe";
+
+import { useCallback, useState } from "react";
+
+import { isGameOver, isWinner } from "../utilities/ticTacToe";
+import { useScore } from "../components/Score";
+
+const createNewTicTacToeBoard = () => Array(9).fill(null);
+
+const fetchWinCheck = async (url: string, data: any) => {
+  const response = await fetch(url, data);
+  const finalData = await response.json();
+  return finalData;
+};
+
+export const useTicTacToe = () => {
+  const { addWinO, addWinX } = useScore();
+  const [board, _setBoard] = useState<TicTacToeBoard>(createNewTicTacToeBoard);
+  const [isX, setIsX] = useState(true);
+  const [winner, setWinner] = useState<BoardSquareValue>(null);
+
+  const [playerNames, setPlayerNames] = useState({});
+
+  const currentTurn: BoardSquareValue = isX ? "X" : "O";
+  const setBoard = (board: any) => {
```

Comment:
Should `any` be `TicTacToeBoard`? Or maybe you’ll remove this function because it’s just `_setBoard`?